libminifi/include/utils/ThreadPool.h

```diff
@@ -384,9 +358,11 @@ class ThreadPool {
   moodycamel::ConcurrentQueue<std::shared_ptr<WorkerThread>> deceased_thread_queue_;
 // worker queue of worker objects
   moodycamel::ConcurrentQueue<Worker<T>> worker_queue_;
-  std::priority_queue<Worker<T>, std::vector<Worker<T>>, WorkerComparator<T>> worker_priority_queue_;
+  std::priority_queue<Worker<T>, std::vector<Worker<T>>, DelayedTaskComparator<T>> delayed_worker_queue_;
```

Comment:
I'd consider marking `Worker<T>`'s move ctor and move assignment operator as `noexcept`. I have bad feelings about relying on the vector exception for move-only not-nothrow-movable types, especially since all exception guarantees are waived.
The only non-static data member in Worker that could in theory throw on move is `std::string` and it will not throw in practice with the default allocator.

Reply:
Done